### PR TITLE
[20.10 backport] Remove local fork of archive/tar package

### PR DESCRIPTION
**Pull request by corhere**
- 20.10 backport of #43185

The hack of forking archive/tar is incompatible with the version of the package in Go 1.20.
```
vendor/archive/tar/common.go:16:2: use of internal package internal/godebug not allowed
```

Backport the proper fix rather than hacking on the hack.

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by neersighted — APPROVED** [review]

**Comment by thaJeztah** [comment]
Linter is complaining; you probably need to cherry-pick https://github.com/moby/moby/commit/65e1adc219f4b1b71f09479aa98be87135a579d5 (part of https://github.com/moby/moby/pull/43765)


```
[2023-10-20T22:21:07.807Z] pkg/archive/archive.go:469:3: SA1019: hdr.Xattrs has been deprecated since Go 1.10: Use PAXRecords instead. (staticcheck)
[2023-10-20T22:21:07.807Z] 		hdr.Xattrs = make(map[string]string)
[2023-10-20T22:21:07.807Z] 		^
[2023-10-20T22:21:07.807Z] pkg/archive/archive.go:470:3: SA1019: hdr.Xattrs has been deprecated since Go 1.10: Use PAXRecords instead. (staticcheck)
[2023-10-20T22:21:07.807Z] 		hdr.Xattrs["security.capability"] = string(capability[:length])
[2023-10-20T22:21:07.807Z] 		^
[2023-10-20T22:21:07.807Z] pkg/archive/archive.go:724:26: SA1019: hdr.Xattrs has been deprecated since Go 1.10: Use PAXRecords instead. (staticcheck)
[2023-10-20T22:21:07.807Z] 	for key, value := range hdr.Xattrs {
[2023-10-20T22:21:07.807Z] 	                        ^
[2023-10-20T22:21:07.807Z] pkg/archive/archive_linux.go:45:7: SA1019: hdr.Xattrs has been deprecated since Go 1.10: Use PAXRecords instead. (staticcheck)
[2023-10-20T22:21:07.807Z] 			if hdr.Xattrs != nil {
[2023-10-20T22:21:07.807Z] 			   ^
[2023-10-20T22:21:07.807Z] pkg/archive/archive_linux.go:46:12: SA1019: hdr.Xattrs has been deprecated since Go 1.10: Use PAXRecords instead. (staticcheck)
[2023-10-20T22:21:07.807Z] 				delete(hdr.Xattrs, "trusted.overlay.opaque")
[2023-10-20T22:21:07.807Z] 				       ^
[2023-10-20T22:21:07.807Z] pkg/tarsum/versioning.go:122:34: SA1019: h.Xattrs has been deprecated since Go 1.10: Use PAXRecords instead. (staticcheck)
[2023-10-20T22:21:07.807Z] 	xAttrKeys := make([]string, len(h.Xattrs))
[2023-10-20T22:21:07.807Z] 	                                ^
[2023-10-20T22:21:07.807Z] pkg/tarsum/versioning.go:123:17: SA1019: h.Xattrs has been deprecated since Go 1.10: Use PAXRecords instead. (staticcheck)
[2023-10-20T22:21:07.807Z] 	for k := range h.Xattrs {
[2023-10-20T22:21:07.807Z] 	               ^
[2023-10-20T22:21:07.807Z] pkg/tarsum/versioning.go:139:56: SA1019: h.Xattrs has been deprecated since Go 1.10: Use PAXRecords instead. (staticcheck)
[2023-10-20T22:21:07.807Z] 		orderedHeaders = append(orderedHeaders, [2]string{k, h.Xattrs[k]})
[2023-10-20T22:21:07.807Z] 		                                                     ^
```

**Comment by corhere** [comment]
The only CI failures are the usual Windows Server hcsshim timeouts. PTAL @thaJeztah

File: hack/vendor.sh [hunk 1]
```diff
@@ -1,28 +0,0 @@
-#!/usr/bin/env bash
```
Comment (thaJeztah) [inline hack/vendor.sh:1]: Looks like you removed a bit too much here; the `hack/vendor.sh` script itself should probably still be there (or Jenkins to be updated);


```
[2023-10-23T17:06:17.709Z] + docker run --rm -t --privileged -v /home/ubuntu/workspace/moby_PR-46695/.git:/go/src/github.com/docker/docker/.git --name docker-pr2 -e DOCKER_EXPERIMENTAL -e DOCKER_GITCOMMIT=d6cbc6b37d407d5bc8573bcfbbcf803b8fcb91e4 -e DOCKER_GRAPHDRIVER -e TEST_FORCE_VALIDATE -e VALIDATE_REPO=https://github.com/moby/moby.git -e VALIDATE_BRANCH=20.10 docker:d6cbc6b37d407d5bc8573bcfbbcf803b8fcb91e4 hack/validate/vendor
[2023-10-23T17:06:17.969Z] hack/validate/vendor: line 14: ./hack/vendor.sh: No such file or directory
script returned exit code 127
```

Comment (corhere) [reply hack/vendor.sh:1]: Whoops, missed that one. Those pesky horizontal scroll bars...

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**